lib/validators/specTransformer.ts

```diff
@@ -1,49 +1,52 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
-import { SchemaObject, SwaggerObject } from "yasway"
-import { entries, StringMap } from "@ts-common/string-map"
-import { FilePosition, getInfo, getRootObjectInfo, getAllDirectives } from "@ts-common/source-map"
-
-/**
- * Transforms the swagger object
- */
-export function transform(spec: SwaggerObject): SwaggerObject {
-  if (!spec.definitions) {
-    return spec
-  }
-
-  for (const [definitionName, definition] of entries(spec.definitions)) {
-    insertSchemaTitle(definition, `/definitions/${definitionName}`)
-
-    if (definition.properties) {
-      for (const [propertyName, property] of entries(definition.properties)) {
-        insertSchemaTitle(
-          property,
-          `/definitions/${definitionName}/properties/${propertyName}`
-        )
-      }
-    }
-  }
-
-  return spec
-}
+import { SchemaObject } from "yasway"
+import { StringMap } from "@ts-common/string-map"
+import {
+  FilePosition, getRootObjectInfo, getAllDirectives, getPath, InfoFunc, getInfoFunc, setInfoFunc
+} from "@ts-common/source-map"
 
 export interface TitleObject {
   readonly position?: FilePosition
   readonly directives?: StringMap<unknown>
   readonly url?: string
+  readonly path?: ReadonlyArray<string | number>
+}
+
+export interface SchemaObjectInfo {
   readonly title: string
+  readonly infoFunc: InfoFunc
+}
+
+export const getSchemaObjectInfo = (model: SchemaObject): SchemaObjectInfo | undefined => {
+  const infoFunc = getInfoFunc(model)
+  if (infoFunc === undefined) {
+    return undefined
+  }
+  const info = infoFunc()
+  return {
+    title: JSON.stringify(
+      {
+        path: getPath(info),
+        position: info.position,
+        url: getRootObjectInfo(info).url,
+        directives: getAllDirectives(model, []),
+      }
+    ),
+    infoFunc
+  }
 }
 
-function insertSchemaTitle(model: SchemaObject, title: string) {
-  const info = getInfo(model)
-  const titleObject: TitleObject = info === undefined ?
-    { title } :
-    {
-      title,
-      position: info.position,
-      url: getRootObjectInfo(info).url,
-      directives: getAllDirectives(model, []),
-    }
-  model.title = JSON.stringify(titleObject)
+export const setSchemaInfo = (
+  model: SchemaObject,
+  info: SchemaObjectInfo | undefined,
+): SchemaObject => {
+  if (info !== undefined) {
+    model.title = info.title
+    setInfoFunc(model, info.infoFunc)
```

Comment:
it's so hard to tell what any of this does when i have to jump into other repos !

Reply:
I've added comment.